java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowWriter.java

```diff
@@ -119,6 +119,53 @@ protected ArrowWriter(VectorSchemaRoot root, DictionaryProvider provider, Writab
     this.schema = new Schema(fields, root.getSchema().getCustomMetadata());
   }
 
+  /**
+   * Note: fields are not closed when the writer is closed.
+   *
+   * @param root               the vectors to write to the output
+   * @param provider           where to find the dictionaries
+   * @param out                the output where to write
+   * @param option             IPC write options
+   * @param compressionFactory Compression codec factory
+   * @param codecType          Compression codec
+   * @param compressionLevel   Compression level
+   */
+  protected ArrowWriter(VectorSchemaRoot root, DictionaryProvider provider, WritableByteChannel out, IpcOption option,
```

Comment:
This duplicates a lot with the previous ctor. Probably we can use one as a delegate or refactor one to accept a created codec?

Reply:
Agreed, can we consolidate?

Reply:
Updated

Reply:
This now breaks compatibility. I don't think that's what Gang was suggesting with a delegate constructor.

Reply:
Oh, I've decided to refactor constructor to add new parameter needed, .... otherwise will be to maintain the constructor as it is now and add a new constructor for new compression_level args needed. Please let me know how to proceed?

Reply:
As already suggested: why not add a new constructor, and have the old ones delegate to the new one? That way we can avoid code duplication.

Reply:
Updated